CommitValidationTest fails on revert commits

Commit: 26cadbb6772aa6ca85d6e95a9bff5d5366ca0430
Message: `Revert "doc: release notes for 10.8.0"`
Pipeline fail: https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=13264&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=795
```
[ERROR] Failures: 
[ERROR]   CommitValidationTest.testCommitMessageHasProperStructure:183 Commit 26cadbb6772aa6ca85d6e95a9bff5d5366ca0430 message: "Revert "doc: release notes for 10.8.0"\n\nThis reverts commit ff8bcc73c3c22161656794c969bb28a8cb09595f.\n" is invalid
Proper commit message should adhere to the following rules:
    1) Must match one of the following patterns:
        ^Issue #\d+: .*$
        ^Pull #\d+: .*$
        ^(minor|config|infra|doc|spelling|dependency|supplemental): .*$
    2) It contains only one line of text
    3) Must not end with a period, space, or tab
    4) Commit message should be less than or equal to 200 characters

The rule broken was: 1
[INFO] 
[ERROR] Tests run: 3931, Failures: 1, Errors: 0, Skipped: 2

```

--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/CommitValidationTest.java
@@ -74,6 +74,8 @@ public class CommitValidationTest {
     private static final String PR_COMMIT_MESSAGE_REGEX_PATTERN = "^Pull #\\d+: .*$";
     private static final String RELEASE_COMMIT_MESSAGE_REGEX_PATTERN =
             "^\\[maven-release-plugin] .*$";
+    private static final String REVERT_COMMIT_MESSAGE_REGEX_PATTERN =
+            "^Revert .*$";
     private static final String OTHER_COMMIT_MESSAGE_REGEX_PATTERN =
             "^(minor|config|infra|doc|spelling|dependency|supplemental): .*$";
 
@@ -156,6 +158,21 @@ public class CommitValidationTest {
     }
 
     @Test
+    public void testRevertCommitMessage() {
+        assertWithMessage("should accept proper revert commit message")
+                .that(validateCommitMessage("Revert \"doc: release notes for 10.8.0\""
+                    + "\nThis reverts commit ff873c3c22161656794c969bb28a8cb09595f.\n"))
+                .isEqualTo(0);
+        assertWithMessage("should accept proper revert commit message")
+                .that(validateCommitMessage("Revert \"doc: release notes for 10.8.0\""))
+                .isEqualTo(0);
+        assertWithMessage("should not accept revert commit message with invalid prefix")
+                .that(validateCommitMessage("This reverts commit "
+                        + "ff873c3c22161656794c969bb28a8cb09595f.\n"))
+                .isEqualTo(1);
+    }
+
+    @Test
     public void testSupplementalPrefix() {
         assertWithMessage("should accept commit message with supplemental prefix")
                 .that(0)
@@ -206,7 +223,11 @@ public class CommitValidationTest {
         final String trimRight = commitMessage.replaceAll("[\\r\\n]+$", "");
         final int result;
 
-        if (!ACCEPTED_COMMIT_MESSAGE_PATTERN.matcher(message).matches()) {
+        if (message.matches(REVERT_COMMIT_MESSAGE_REGEX_PATTERN)) {
+            // revert commits are excluded from validation
+            result = 0;
+        }
+        else if (!ACCEPTED_COMMIT_MESSAGE_PATTERN.matcher(message).matches()) {
             // improper prefix
             result = 1;
         }

